resolver/resolver.go

```diff
@@ -467,29 +472,28 @@ func truncate(m *dns.Msg, udp bool) *dns.Msg {
 
 	furtherTruncation := m.Len() > max
 	m.Truncated = m.Truncated || furtherTruncation
-
+	if !setTruncateBit {
```

Comment:
This chunk of `m.Truncated` setting also struck me as kind of strange looking, but I couldn't think of a better way to express it off the top of my head.